.github/workflows/docusaurus.yml

```diff
@@ -2,30 +2,34 @@ name: Publish Docusaurus to staging
 
 on:
   push:
-    branches:
-      - develop
```

Comment:
If you remove it, whenever a push is done to any branch, it will be run.
Why don't we filter it to only run on `develop` and `main` branches?
We follow a kind of "git-flow" approach, so `develop` branch should be stable and keep the source of any "snapshot" version (so docs can be published to staging), and `main` will contains any code that has been already released, so docs can be published to production.
If we don't filter this job by these branches, we can get some "wrong docs" on staging. Not only that, we will be spending CI time when it is not needed

Reply:
It will be triggered only when the PR changes relates to `'docusaurus/**'` and `'.github/workflows/docusaurus.yml'` paths

Reply:
You can be working in a branch about some docs that is not ready to be merged to `develop` yet, and this workflow will be run, publishing docs that is not ready on Staging

Reply:
Fair enough, but isn't this a good way to test the docs on your PR? I don't have strong opinion on this, but this is how it currently works on React, RN and iOS 🤔

Reply:
Kind of on the same page as @testableapple.

I think we have 2 options:
* tailor the action towards your workflow
* keep it as close to the other repos as possible

Since each team has a different flow, we won't have the perfect one-size-fits-all solution.
I don't have a strong preference, and of course you @JcMinarro should be the one deciding on something!

I just don't hate that you can test the docs even when you have a WIP branch on `staging`.

Reply:
Can we move this forward or are there still concerns, @JcMinarro ?